### PR TITLE
Nbhd evo

### PR DESCRIPTION
**Pull request by riveSunder**

### CODE REVIEW

File: yuca/wrappers/glider_wrapper.py [hunk 1]
```diff
@@ -94,37 +97,44 @@ def meta_step(self, action):
...
                 y_displacement_0 = y_displacement_1.clone()
 
+
             dgrid = action - old_grid
             old_grid = 1.0  * action
 
             if mean_grid == 0.0 or dgrid.max() <= 0.00001:
                 break
 
-        if step == 0:
```
Comment (riveSunder) [inline yuca/wrappers/glider_wrapper.py:120]: Clean up commented code

File: yuca/wrappers/glider_wrapper.py [hunk 2]
```diff
@@ -94,37 +97,44 @@ def meta_step(self, action):
...
-            # patterns that die out are substantially penalized 
-            reward -= 100
+#        if step == 0:
+#            # empty patterns are extremely penalized 
+#            reward -= 10000
+#        if mean_grid == 0.0 or dgrid.max() <= 0.00001:
+#            # patterns that die out are substantially penalized 
+#            reward -= 100
         
-        reward -= (self.ca_steps - step)
```
Comment (riveSunder) [inline yuca/wrappers/glider_wrapper.py:127]: It _seems_ like we should be able to leave this part out, because agents will get more rewards for longer run times naturally. In practice there seems to be a fine line between inadvertently selecting for long-lived static patterns and transient mobile ones.

File: yuca/wrappers/glider_wrapper.py [hunk 3]
```diff
@@ -138,6 +148,9 @@ def init_displacement_grid(self):
         self.x_grid = torch.tensor(xx).reshape(1, 1, self.dim, self.dim)
         self.y_grid = torch.tensor(yy).reshape(1, 1, self.dim, self.dim)
 
+        self.x_grid = self.x_grid / self.ca.kernel_radius
+        self.y_grid = self.y_grid / self.ca.kernel_radius
```
Comment (riveSunder) [inline yuca/wrappers/glider_wrapper.py:152]: Mobiity is normalized w.r.t. the kernel radius, _i.e._ the maximum speed for information to travel in the simulation (aka the speed of light)

**Review by riveSunder — COMMENTED** [review]
Remove dead code in glider wrapper step function. 
Then OK to merge.

File: coverage.txt [hunk 4]
```diff
@@ -3,20 +3,20 @@ Name                               Stmts   Miss  Cover
...
+yuca/utils.py                        131      8    94%
 yuca/wrappers/__init__.py              3      0   100%
-yuca/wrappers/glider_wrapper.py       88     10    89%
-yuca/wrappers/halting_wrapper.py     177     28    84%
+yuca/wrappers/glider_wrapper.py       90     12    87%
+yuca/wrappers/halting_wrapper.py     182     28    85%
 yuca/wrappers/random_wrapper.py       16      4    75%
 ------------------------------------------------------
-TOTAL                               1881    250    87%
+TOTAL                               1922    259    87%
```
Comment (riveSunder) [inline coverage.txt:22]: ✓
